test/functional/homepage_controller_test.rb

```diff
@@ -5,6 +5,86 @@ class HomepageControllerTest < ActionController::TestCase
     login_as_stub_user
   end
 
+  context "#confirm_destroy" do
+    should "render confirm destroy page for draft edition" do
+      popular_links = FactoryBot.create(:popular_links, state: "draft")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_template "homepage/popular_links/confirm_destroy"
+    end
+
+    should "redirect with error message if edition is published" do
+      popular_links = FactoryBot.create(:popular_links, state: "published")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_redirected_to show_popular_links_path
+      assert_equal "Can't delete an already published edition. Please create a new edition to make changes.", flash[:danger]
+    end
+  end
+
+  context "destroy" do
+
+    context "edition is draft" do
+      setup do
+        @popular_links = FactoryBot.create(:popular_links, state: "draft")
+        get :confirm_destroy, params: { id: @popular_links.id }
+      end
+
+      should "delete the popular links from the database and display success message" do
+        assert_equal 1, PopularLinksEdition.count
+
+        delete :destroy, params: { id: @popular_links.id }
+
+        assert_equal 0, PopularLinksEdition.count
+        assert_equal "Popular links draft deleted.", flash[:success]
+      end
+
+      should "redirect to show path on success" do
+        delete :destroy, params: { id: @popular_links.id }
+
+        assert_redirected_to show_popular_links_path
+      end
+    end
+
+    context "edition is published" do
+      setup do
+        @popular_links = FactoryBot.create(:popular_links, state: "published", version_number: "2")
+      end
+
+      should "show 'cant delete published edition' when trying to delete a published edition" do
+        delete :destroy, params: { id: @popular_links.id }
+
+        assert_equal "Can't delete an already published edition. Please create a new edition to make changes.", flash[:danger]
+      end
+    end
+
+    context "database errors" do
+      setup do
+        @popular_links = FactoryBot.create(:popular_links, state: "draft")
+      end
+
+      should "redirect to show page with 'application error' if delete from database fails" do
```

Comment:
test on line 68 and 77 seem to be doing one thing and can be a single test.
or you can have two separate test one for redirect and one for alert message

Also, can we bring the whole context for "confirm destroy" to after publish context block.